authzed/api/v1/experimental_service.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+package authzed.api.v1;
+
+option go_package = "github.com/authzed/authzed-go/proto/authzed/api/v1";
+option java_package = "com.authzed.api.v1";
+
+import "google/api/annotations.proto";
+import "validate/validate.proto";
+
+import "authzed/api/v1/core.proto";
+
+// ExperimentalService exposes a number of APIs that are currently being
+// prototyped and tested for future inclusion in the stable API.
+service ExperimentalService {
+
+  // BulkLoadRelationships is a faster path to writing a large number of
+  // relationships at once. It is both batched and streaming. For maximum
+  // performance, the caller should attempt to write relationships in as close
+  // to relationship sort order as possible: (resource.object_type,
+  // resource.object_id, relation, subject.object.object_type,
+  // subject.object.object_id, subject.optional_relation)
+  // 
+  // EXPERIMENTAL
+  rpc BulkLoadRelationships(stream BulkLoadRelationshipsRequest)
```

Comment:
I wonder if we should put the `Experimental` in the API name itself too, for the combined clients

Reply:
The clients could do this themselves

Reply:
Putting the name in the API sucks for migrating later...

Reply:
That's actually pain I want to cause, for exactly this reason: it ensures that callers aren't using it for real and, if they are, they *know* to look again once renamed

Reply:
In `authzed-go` you have to initialize your client by calling `NewClientWithExperimentalAPIs`. Pretty clear...

Reply:
Yeah, but in `authzed-node`, while we'd likely make you have to give us the "withExperimental" flag, once that's passed in, the *client* is the same type (since it is mixed). That means if once we move out of experimental, we change the semantics of the call, without changing the call signature, it can break

Reply:
In go I made a different type client with different mixins. Is that not a scalable approach to other languages?

Reply:
We could look into doing so, but it would change how we do clients in authzed-node

Reply:
I'd prefer to not reengineer the authzed-node client and break it for existing consumers, so if we do so, we should do that for V2. For now, I'd prefer we either change the method names here or just recognize the risk with having the methods not being marked as experimental at the call site

Reply:
Nothing would be broken for existing clients. In go I provide an alternate constructor that gives back an extended type:

```go

// Client represents an open connection to Authzed.
//
// Clients are backed by a gRPC client and as such are thread-safe.
type Client struct {
	v1.SchemaServiceClient
	v1.PermissionsServiceClient
	v1.WatchServiceClient
}

// ClientWithExperimental represents and open connection to Authzed with
// experimental services available.
//
// Clients are backed by a gRPC client and as such are thread-safe.
type ClientWithExperimental struct {
	Client

	v1.ExperimentalServiceClient
}

// NewClient initializes a brand new client for interacting with Authzed.
func NewClient(endpoint string, opts ...grpc.DialOption) (*Client, error) {
	conn, err := newConn(endpoint, opts...)
	if err != nil {
		return nil, err
	}

	return &Client{
		v1.NewSchemaServiceClient(conn),
		v1.NewPermissionsServiceClient(conn),
		v1.NewWatchServiceClient(conn),
	}, nil
}

// NewClientWithExperimentalAPIs initializes a brand new client for interacting
// with Authzed.
func NewClientWithExperimentalAPIs(endpoint string, opts ...grpc.DialOption) (*ClientWithExperimental, error) {
	conn, err := newConn(endpoint, opts...)
	if err != nil {
		return nil, err
	}

	return &ClientWithExperimental{
		Client{
			v1.NewSchemaServiceClient(conn),
			v1.NewPermissionsServiceClient(conn),
			v1.NewWatchServiceClient(conn),
		},
		v1.NewExperimentalServiceClient(conn),
	}, nil
}

func newConn(endpoint string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
	return grpc.Dial(
		stringz.DefaultEmpty(endpoint, "grpc.authzed.com:443"),
		opts...,
	)
}

```

Reply:
Yes, it is similar in JS, but if we changes the semantics of the method after the fact, then it can break without the caller realizing

Reply:
I'm going to LGTM but we'll have to be very careful with changing how the API works